File: CS50 2019/C/pset1/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    long n;
    do
    {
        n = get_long("Number: ");
    }
    while(n < 0);
    
    int sum1 = 0;
    int sum2 = 0;

    for (long l = n; l > 0; l /= 100)
    {
        sum1 += (l % 10);
    }
    
    for (long l = n / 10; l > 0; l /= 100)
    {
        if ((l % 10) * 2 > 9)
        {
            long a = ((l % 10) * 2) % 10;
            long b = ((l % 10) * 2) / 10;
            sum2 += a + b;
        }
        else
        {
            sum2 += (l % 10) * 2;
        }
    }
    int sum = sum1 + sum2;
    
    if (sum % 10 == 0)
    {
        if ((n >= 340000000000000L && n < 350000000000000L) || (n >= 370000000000000L && n < 380000000000000L))
         {
             printf("AMEX\n");
         }
         else if (n >= 5100000000000000L && n < 5600000000000000L)
         {
             printf("MASTERCARD\n");
         }
         else if ((n >= 4000000000000L && n < 5000000000000L) || (n >= 4000000000000000L && n < 5000000000000000L))
         {
             printf("VISA\n");
         }
         else
         {
             printf("INVALID\n");
         }
    }
    else 
    {
        printf("INVALID\n");
    }
}

File: CS50 2019/Python/pset6/credit.py
from cs50 import get_int

def main():
    while True:
        number = get_int("Number: ")
        if number > 0:
            break
        
    number1 = number
    length = len(str(number))
    sum1 = 0
    sum2 = 0
    while number > 0:
        sum1 += (number % 10)
        tmp = number // 10
        if (tmp % 10) * 2 > 9:
            a = ((tmp % 10) * 2) % 10
            b = ((tmp % 10) * 2) // 10
            sum2 += a + b
        else:
            sum2 += (tmp % 10) * 2
        
        number //= 100
    
    summa = sum1 + sum2
    
    if summa % 10 == 0:
        if (number1 >= 340000000000000 and number1 < 350000000000000) or (number1 >= 370000000000000 and number1 < 380000000000000):
            print("AMEX")
        elif number1 >= 5100000000000000 and number1 < 5600000000000000:
            print("MASTERCARD")
        elif (number1 >= 4000000000000 and number1 < 5000000000000) or (number1 >= 4000000000000000 and number1 < 5000000000000000):
            print("VISA")
        else:
            print("INVALID")
    else:
        print("INVALID")
    
if __name__ == "__main__":
    main()